src/test/java/test/failedreporter/FailedReporterTest.java

```diff
@@ -31,30 +42,106 @@ public void tearDown() {
-        "<include name=\"%s\"" + "\"/>");   }
+    testFailedReporter(new String[] { "f1", "f2" }, "<include name=\"%s\"/>",
+        FailedReporterSampleTest.class);
+  }
 
   @Test
   public void failedMethodWithDataProviderShouldHaveInvocationNumbers() throws IOException {
-    testFailedReporter(FailedReporter2SampleTest.class, new String[] { "f1" },
-        "<include invocationNumbers=\"1\" name=\"%s\"" + "\"/>");
+    testFailedReporter(new String[] { "f1" }, "<include name=\"%s\" invocation-numbers=\"1\"/>",
+        FailedReporter2SampleTest.class);
+  }
+
+  @Test(description = "github-1297")
+  public void testExclusionOfPassedConfigs() {
+    triggerTest(AllPassSample.class, FailureSample.class);
+    String[] substitutions = new String[] {AllPassSample.class.getName()};
+    runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">",0);
+    substitutions = new String[] {"newTest2", "beforeClassFailureSample","afterClassFailureSample"};
+    runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>",1);
   }
 
-  private void testFailedReporter(Class<?> cls, String[] expectedMethods, String expectedLine) {
+  @Test(description = "github-1297")
+  public void testExclusionOfPassedConfigsAmidstInheritance() {
+    triggerTest(InheritanceFailureSample.class, InheritancePassSample.class);
+    String[] substitutions = new String[] {InheritancePassSample.class.getName()};
+    runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">", 0);
+    substitutions = new String[] {"newTest2", "baseBeforeTest","baseAfterClass","baseBeforeClass","baseBeforeMethod"};
+    runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>",1);
+  }
+
+  @Test(description = "github-1297")
+  public void testExclusionOfPassedConfigsInvolvingGroupsAtTestLevel() {
+    triggerTest(GroupsSampleBase.class.getPackage().getName(),true);
+    String[] substitutions = new String[] {GroupsPassSample.class.getName()};
+    runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">",0);
+    substitutions = new String[] {"baseBeforeTest", "baseBeforeClassAlwaysRun","newTest2"};
+    runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>",1);
+  }
+
+  @Test(description = "github-1297")
+  public void testExclusionOfPassedConfigsInvolvingGroupsAtSuiteLevel() {
+    triggerTest(GroupsSampleBase.class.getPackage().getName(),false);
+    String[] substitutions = new String[] {GroupsPassSample.class.getName()};
+    runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">",0);
+    substitutions = new String[] {"baseBeforeTest", "baseBeforeClassAlwaysRun","newTest2"};
+    runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>",1);
+  }
+
+  private void triggerTest(String packageName, boolean applyGroupSelectionAtTest) {
     TestNG tng = new TestNG();
     tng.setVerbose(0);
-    tng.setTestClasses(new Class[] { cls });
+    final XmlSuite suite = new XmlSuite();
```

Comment:
Use helper methods from `SimpleBaseTest`. Maybe you'll have to update some legacy tests too.